OpenRA.Game/Traits/World/ScreenMap.cs

```diff
@@ -67,6 +70,16 @@ public void Add(Player viewer, FrozenActor fa)
 			partitionedFrozenActors[viewer].Add(fa, FrozenActorBounds(fa));
 		}
 
+		public void Add(IEffect effect, WPos position)
```

Comment:
Effects can (and do: projectiles) move.  The effects themselves will need to tell the screenmap when they are added, moved, or removed from the world in the same way that actors do.